src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/DocumentBasedFixAllProvider.cs

```diff
@@ -123,14 +123,14 @@ private static async Task<ImmutableArray<Diagnostic>> DetermineDiagnosticsAsync(
                 }
                 else
                 {
-                    var diagnostics = ImmutableArray<Diagnostic>.Empty;
+                    using var _1 = ArrayBuilder<Diagnostic>.GetInstance(out var diagnostics);
                     foreach (var fixAllSpan in fixAllContext.State.FixAllSpans)
                     {
-                        diagnostics = diagnostics.AddRange(await fixAllContext.GetDocumentSpanDiagnosticsAsync(
-                            fixAllContext.Document, fixAllSpan).ConfigureAwait(false));
+                        var spanDiagnostics = await fixAllContext.GetDocumentSpanDiagnosticsAsync(fixAllContext.Document, fixAllSpan).ConfigureAwait(false);
+                        diagnostics.AddRange(spanDiagnostics);
                     }
 
-                    return diagnostics;
+                    return diagnostics.ToImmutable();
```

Comment:
does this code run oop?  if so, consider getting doc diagnostics in parallel.

Reply:
This doesn't run in OOP